tests/geometry/surfaces/testIntegralInvariantCovarianceEstimator.cpp

```diff
@@ -203,22 +209,40 @@ bool testPrincipalCurvatures3d( double h )
 
   MyIICurvatureFunctor curvatureFunctor;
   curvatureFunctor.init( h, re );
-
   MyIICurvatureEstimator curvatureEstimator( curvatureFunctor );
   curvatureEstimator.attach( K, dshape );
   curvatureEstimator.setParams( re/h );
   curvatureEstimator.init( h, ibegin, iend );
 
+  MyIICurvatureFunctorTensor curvatureFunctorTensor;
+  curvatureFunctorTensor.init( h, re );
+  MyIICurvatureEstimatorTensor curvatureEstimatorTensor( curvatureFunctorTensor );
+  curvatureEstimatorTensor.attach( K, dshape );
+  curvatureEstimatorTensor.setParams( re/h );
+  curvatureEstimatorTensor.init( h, ibegin, iend );
+  
   trace.endBlock();
 
   trace.beginBlock( "Curvature estimator evaluation ...");
 
   std::vector< Value > results;
   std::back_insert_iterator< std::vector< Value > > resultsIt( results );
   curvatureEstimator.eval( ibegin, iend, resultsIt );
+  trace.endBlock();
 
+  
+  
+  trace.beginBlock( "Checking CurvaturesAndDirections functor");
+  Value val = curvatureEstimator.eval( surf.begin() );
+  ValueTensor valTensor = curvatureEstimatorTensor.eval( surf.begin() );
+  bool ok = (val.first ==  std::get<0>(valTensor));
+  ok &= (val.second ==  std::get<1>(valTensor));
+  if (!ok)
+    trace.error()<< "Error comparing principal curvatures between two different functors."<<std::endl;
+  
```

Comment:
Shouldn't we test the estimation values also?

Reply:
No need I think. The first functor is already tested. We just make sure that the approximations coincide

Reply:
I've added another test to make sure that the Shortcuts are fine.

Reply:
Ok I see now, thx!